judge/models/problem.py

```diff
@@ -451,6 +451,25 @@ def save(self, *args, **kwargs):
 
     save.alters_data = True
 
+    def can_vote(self, user):
+        if not user.is_authenticated:
+            return False
+
+        # If the user is in contest, nothing should be shown.
+        if user.profile.current_contest:
+            return False
+
+        # If the user is not allowed to vote
+        if user.profile.is_unlisted or user.profile.is_banned_problem_voting:
+            return False
+
+        # If the user is banned from submitting to the problem.
+        if self.banned_users.filter(pk=user.pk).exists():
+            return False
+
+        # If the user has a full AC submission to the problem (solved the problem).
+        return self.submission_set.filter(user=user.profile, result='AC', points=F('problem__points')).exists()
```

Comment:
This is a duplicate check from the `submission.can_see_detail` check. Let's refactor this.